src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/Navigation.java

```diff
@@ -169,35 +134,28 @@ private void addMainNavLinks(Content target) {
         switch (documentedPage) {
```

Comment:
Maybe not now, but I have long felt this big switch statement could be better served as a virtual abstract method, with suitable implementations in each kind of page.